### DISCUSSION TITLE
Stryker errors when the Since option is set

### DISCUSSION THREAD

**Discussion by JosePedroRoriz** [stryker-mutator/stryker-net#2451] [Q&A]
First of all let me say thank you for creating this amazing tool.

I am currently in the process of having stryker run on our pipelines and managed to get it to the point where it does a full mutation. My issue now is I only want to mutate changes done at a PR level. From reading the stryker .net configuration documentation it seems to suggest using the Since option with the current branch name. 
Having done that stryker starts running and fails with an error message from LibGit2Sharp.NotFoundException "requested file is a directory". I built a small spike solution using the LibGit2Sharp solution where i hard coded my git paths (everything seemed correct) and it fails when it tried to run git_patch_from_diff. From here i ran a git diff where it showed the changes i expected and a git diff>someStuff.patch where it also had what i expected. From running stryker locally this happens in GitDiffProvider.cs inside the ScanDiff method.
So my question here is, is the problem with my stryker.config? Or is it something else?

My stryker config looks like this:

{
  "stryker-config": {
    "project": "My.Proj.Services",
    "report-file-name": "My.Proj.Services",
    "since": {
      "enabled": true,
      "target": "MyBranch"
    },
    "concurrency": 8,
    "reporters": [
      "progress",
      "html"
    ],
    "thresholds": {
      "high": 80,
      "low": 80,
      "break": 80
    },
    "mutation-level": "Standard",
    "ignore-methods": [
      "ToString",
      "Console.Write*"
    ],
    "mutate": [
      "!Program.cs",
      "!Startup.cs"
    ]
  }
}

The change i did in MyBranch was adding a simple method to a service in hopes of seeing a mutation score of 0 for that service.

Please let me know if more info is needed to help diagnose why this is happening.

Kind regards,

Pedro

**Comment by rouke-broersma** [comment]
Which ci tool do you use? Might be that they do some kind of shallow checkout that doesn't contain the info we need in scandiff.

**Reply by JosePedroRoriz** [reply]
I apologize, that error shows up when I run stryker locally, so no CI involved. I just open a powershell console to my unit test proj and typed dotnet stryker.

I have recently tried running it on Github Actions (the CI tool we use), and a different error seems to occur.  If i leave the Since option as is, it returns an error saying "Could not locate git repository. Unable to determine git diff to filter mutants. Did you run inside a git repo?  If not please disable the 'since' feature. 
So I commented out the Since feature and added baseline + project-info, so my config looks like this now:

{
"stryker-config": {
"project": "My.Proj.Services",
"report-file-name": "My.Proj.Services",
"concurrency": 8,
"project-info":{
"name": "github.com/My.Proj.Services",
"version": "MyBranch"
},
"baseline":{
"fallback-version": "main",
"enabled":true
}
"reporters": [
"progress",
"html"
],
"thresholds": {
"high": 80,
"low": 80,
"break": 80
},
"mutation-level": "Standard",
"ignore-methods": [
"ToString",
"Console.Write*"
],
"mutate": [
"!Program.cs",
"!Startup.cs"
]
}
}

While also passing --with-baseline:MyBranch in the command line.
So it looks like this: dotnet stryker --with-baseline:MyBranch.

But I am still getting the same error. is this due to me forgetting to add something to the config or something else i need to check with our pipeline? Is there support for running it with Github actions?

Also, is it possible to run stryker locally and get a mutation report based off the differences between main and your current branch?
 
Kind regards,
 
 Pedro

**Reply by rouke-broersma** [reply]
Baseline currently implies the Since feature as well, so it is not currently possible to run Baseline without Since. If Since does not work, Baseline is also not going to work. This is something I would like to change when I have enough time.

That said this error is unexpected, are you sure you're actually running in a full git repo?

**Reply by JosePedroRoriz** [reply]
When running locally yes.  When running in github actions the runner VM  does a checkout so it acts if it's running locally as well.  Our current settings for github actions is to never checkout the full git history and our fetch depth is set to 1.

If you need me to find more details let me know.
